### DISCUSSION TITLE
Appium UiAutomator2 + WebDriverIO clearValue not working on a hybrid app

### DISCUSSION THREAD

**Discussion by SiNONiMiTY** [appium/appium#18640] [Q&A]
Title says, I am using JS to create tests. It went smoothly until I tried using the clearValue() method on an input field. It does not clear the pre-filled input.

Tried this approach for the said element
```
await inputField.click()
await inputField.clearValue()
await inputField.setValue( 'TEST' )
```

Any ideas or anything that I am missing.

```
WebDriverIO: 8.10.2
Appium: 2.0.0-beta.66
```

cross posted in WebDriverIO
https://github.com/webdriverio/webdriverio/discussions/10371

**Comment by jlipps** [comment]
Please paste your complete Appium log

**Reply by SiNONiMiTY** [reply]
Hi @jlipps, I found a workaround to this issue by just clicking the element and issuing Ctrl + A + Backspace in the .keys() method.

Will get back to posting the logs here. Thank you.

**Reply by SiNONiMiTY** [reply]
@jlipps here is the log 
[appium.log](https://github.com/appium/appium/files/11483891/appium.log)

What I did here is
.addValue( '+00000000000' )
.clearValue()
.getValue()

You can see on the logs that the .getValue() method still returned the added value.

**Reply by jlipps** [reply]
the `clear` command completes successfully according to the driver. is there no other element (like a containing or contained element) which might be a better target for the command? otherwise, maybe @mykola-mokhnach has some ideas based on how the clear command is implemented in the driver.

**Reply by mykola-mokhnach** [reply]
I don't have any other ideas there. Sounds like UiAutomator bug. What is actually visible in the UI? Does the element value actually gets cleared? If yes, then it probably needs to be located again to avoid possible caching?

**Reply by SiNONiMiTY** [reply]
@mykola-mokhnach the value '+00000000000' is still visible in the UI as if it was not cleared. That is why I implemented the .keys() workaround. Maybe the clear command needs to be double checked on the driver.

**Reply by mykola-mokhnach** [reply]
in such case this must be a UiAutomator's framework issue. Internally nothing else except of the `clear` API is called on the element. 

You could probably try to workaround the issue by double clicking the element content and deleting it all with backspace.
